src/2d/shallow/surge/model_storm_module.f90

```diff
@@ -540,15 +674,1069 @@ subroutine set_CLE_fields(maux, mbc, mx, my, xlower, ylower,    &
 
         ! Storm description, need in out here since we may update the storm
         ! if at next time point
-        type(model_storm_type), intent(inout) :: storm
+        type(model_storm_type), intent(in out) :: storm
 
         ! Array storing wind and presure field
         integer, intent(in) :: wind_index, pressure_index
         real(kind=8), intent(inout) :: aux(maux,1-mbc:mx+mbc,1-mbc:my+mbc)
 
-        print *, "This model has not yet been implemented!"
-        stop
+        ! Local storage
+        real(kind=8) :: x, y, r, theta, sloc(2) 
+        real(kind=8) :: f, mwr, mws, Pc, Pa, dp, wind, tv(2), radius 
+        real(kind=8) :: mod_mws, trans_speed, ramp 
+        integer :: i,j 
+        
+        ! Variables for CLE model calculations
+        integer, PARAMETER :: res=2000
+        real(kind=8), save :: v_vec(1:res), p_vec(1:res)
+        real(kind=8), save :: m_out_vec(1:res)
+        real(kind=8), save :: last_time=-1
+        real(kind=8), save :: dr, r_a, r_0
+        integer :: io_status
+        integer, save :: in_res, out_res
+
+        
+        ! Additional local storage for CLE Model
+        real(kind=8) :: v, next_loc(2)
+        integer :: r_index
+        real(kind=8) :: r_factor, t_factor
+
+        ! Output the pressure and velocity fields.  
+        ! Determines the radial wind ONLY IF called at a new time. 
+
+        !First, interpolate all of the relevant tracked storm parameters.
+        call get_cle_storm_data(t,storm,sloc,tv,mwr,mws,Pc,radius)
+
+        ! Other quantities of interest 
+        Pa = ambient_pressure 
+
+        ! Additional quantities of interest 
+        i = storm_index(t, storm)
+        f = coriolis(sloc(2))
+        tv = storm%velocity(:, i)
```

Comment:
@hudaquresh @mandli just making sure I understand whats going on here b/c we had previously included the CLE model but had the assignments to `tv` through `mwr` commented out. What's the rationale behind overwriting the values obtained in `get_cle_storm_data` above with new values here? 

Also I think `Pa` is defined twice.

Reply:
@hudaquresh I think this follows the previous Holland model?